crates/holochain_state/src/query.rs

```diff
@@ -294,6 +339,39 @@ impl<'stmt> Store for Txn<'stmt, '_> {
             AnyDht::Header => self.get_exact_element(&hash.clone().into()),
         }
     }
+
+    fn get_public_or_authored_element(
+        &self,
+        hash: &AnyDhtHash,
+        author: Option<&AgentPubKey>,
+    ) -> StateQueryResult<Option<Element>> {
+        match *hash.hash_type() {
+            // Try to get a public element.
+            AnyDht::Entry => match self.get_any_public_element(&hash.clone().into())? {
+                Some(el) => Ok(Some(el)),
+                None => match author {
+                    // If there are none try to get a private authored element.
+                    Some(author) => self.get_any_authored_element(&hash.clone().into(), author),
+                    // If there are no private authored elements then try to get any element and
+                    // remove the entry.
+                    None => Ok(self
+                        .get_any_element(&hash.clone().into())?
+                        .map(|el| Element::new(el.into_inner().0, None))),
```

Comment:
This case doesn't match with the name of the function... isn't this returning something that is neither a public element nor an authored element?

Reply:
Yeh maybe the function needs a better name. It's still valid to get an element for someone elses private entry as long as the entry has been removed (so it's just the header)